### PR TITLE
Limit Flutter Icon customizations to Flutter projects.

### PR DESCRIPTION
**Pull request by pq**
cc @alexander-doroshko @devoncarew

### CODE REVIEW

File: src/io/flutter/project/FlutterIconProvider.java [hunk 1]
```diff
@@ -20,7 +21,7 @@
 public class FlutterIconProvider extends IconProvider {
   @Nullable
   public Icon getIcon(@NotNull final PsiElement element, @Iconable.IconFlags final int flags) {
-    if (element instanceof PsiDirectory) {
+    if (element instanceof PsiDirectory && FlutterModuleUtils.hasFlutterModule(element.getProject())) {
```
Comment (alexander-doroshko) [inline src/io/flutter/project/FlutterIconProvider.java]: Is it possible to detect flutter module just by looking files / folders around?
`hasFlutterModule()` looks too heavy, it even may result in reading a file (`declaresFlutterDependency()`).

BTW is `if (!PlatformUtils.isIntelliJ()) ` branch needed in `hasFlutterModule()`? I thought `FlutterModuleType` used in small IDEs as well. (If it's not the case it means that you can't work with the same project in different IDEs. I hope it;s not the case).

Comment (pq) [inline src/io/flutter/project/FlutterIconProvider.java]: I see your concern.  For this I think checking for a declared Flutter module should suffice.  Updating.

**Comment by pq** [comment]
OK.  Updated to prefer:

    ModuleUtil.hasModulesOfType(..., FlutterModuleType.getInstance())

Let me know what you think.

**Comment by alexander-doroshko** [comment]
Well, the best, of course, is to check that this concrete folder is the flutter project root. You should know better how to check it. I guess it means checking this concrete `pubspec.yaml` file for flutterness which is done in `FlutterModuleUtils#declaresFlutterDependency()`. However that method contains `file.contentsToByteArray()` which I'm not sure about performance, it might be using cached content or not, I don't know. If not - then I'd recommend not to use it here. As an option, you can add your own caching to `declaresFlutterDependency()` to be sure. For example we have similar caching in `PubspecYamlUtil#getPubspecYamlInfo`.

Side note: do you agree that the second part of `hasFlutterModule()` is just a time-consuming noop?

Sorry for too many words. I always keep in mind that there may be thousands of modules of different types, that each module may contain more than one Dart project, that Dart project root may be deeper that a content root, etc., etc. Probably you have many places in the code where you implicitly assume 'standard' flutter project structure and that people do not mix Dart and Flutter projects in one IntelliJ project. Then ok, this will be one more place. Updating and reusing `hasFlutterModule()` might make sense anyway.

Feel free to ignore everything, it's a minor comment.

**Comment by devoncarew** [comment]
The updated PR lgtm.

The 2nd part of `hasFlutterModule()` is there to support the small IDE case, where we don't have control over the module type. As per https://github.com/flutter/flutter-intellij/issues/748, I think we should eventually move to not use a Flutter specific module type; this would let people work with flutter projects even w/o the Flutter plugin installed (just the Dart plugin).

**Pull request merged**